core/src/main/kotlin/ch/ergon/dope/resolvable/clause/SelectClause.kt

```diff
@@ -0,0 +1,97 @@
+package ch.ergon.dope.resolvable.clause
+
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.expression.Expression
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.Field
+import ch.ergon.dope.resolvable.expression.unaliased.type.toNumberType
+import ch.ergon.dope.resolvable.formatToQueryString
+import ch.ergon.dope.validtype.BooleanType
+import ch.ergon.dope.validtype.NumberType
+import ch.ergon.dope.validtype.StringType
+import ch.ergon.dope.validtype.ValidType
+
+interface IOffsetClause : Clause
+
+interface ILimitClause : IOffsetClause {
+    fun offset(numberExpression: TypeExpression<NumberType>): OffsetClause = OffsetClause(numberExpression, this)
+}
+
+enum class OrderByType(val type: String) {
+    ASC("ASC"),
+    DESC("DESC"),
+}
+
+interface IOrderByClause : ILimitClause {
+    fun limit(numberExpression: TypeExpression<NumberType>): LimitClause = LimitClause(numberExpression, this)
+
+    fun limit(number: Number): LimitClause = limit(number.toNumberType())
+}
+
+interface IGroupByClause : IOrderByClause {
+    fun orderBy(stringField: Field<StringType>): OrderByClause = OrderByClause(stringField, this)
+
+    fun orderBy(stringField: Field<StringType>, orderByType: OrderByType): OrderByTypeClause = OrderByTypeClause(stringField, orderByType, this)
+}
+
+interface IWhereClause : IGroupByClause {
+    fun groupBy(field: Field<out ValidType>, vararg fields: Field<out ValidType>): GroupByClause = GroupByClause(field, *fields, parent = this)
+}
+
+interface Fromable : Resolvable
+
+interface IFromClause : IWhereClause {
+    fun where(whereExpression: TypeExpression<BooleanType>) = WhereClause(whereExpression, this)
```

Comment:
isn't it cleaner if we also provide the join methods here. Such that we are able to write something like 

```
.select(...)
.from(...)
.join(...)
.join(...)
.where(...)
...
```

Reply:
Then I think we can remove the Fromable-Interface and use _Bucket_ directly

Reply:
done